io-reactor/src/reactor.rs

```diff
@@ -269,13 +270,17 @@ impl<H: Handler, P: Poll> Runtime<H, P> {
             if count > 0 {
                 self.handle_events(instant);
             }
+            // cloudhead: We should only check this if the waker woke us up.
+            // cloudhead: It would be good to use `try_iter` here.
```

Comment:
This doesn't work b/c of rust borrow checker, which is stupid enough to see a "double mutable borrow" in that case. That was the reason I didn't use it originally